inference/cluster_store.py

```diff
@@ -85,10 +86,21 @@ def build_agglomerative_clusters(self, c):
         # TODO
         pass
 
-    # Function to calculate the average similarity of current clusters 
+    # Function to calculate the average similarity of current clusters
```

Comment:
I think there may have been a miscommunication. This function is meant to calculate the average similarity of the objects inside the clusters to test how well clustering worked on certain features. 

pseudo code:
```python
similarity = 0.0
for cluster in clusters:
    cluster_similarity = 0.0
    for obj1 in cluster:
         for obj2 in cluster:
              cluster_similarity += cosine_similarity(obj1,obj2)
    similarity += cluster_similarity/len(cluster)
avg_similarity = similarity/total_clusters
```